src/include/simeng/arch/aarch64/helpers/auxiliaryFunctions.hh

```diff
@@ -265,16 +265,22 @@ class AuxFunc {
   static uint16_t sveGetPattern(const std::string operandStr,
                                 const uint8_t esize, const uint16_t VL_) {
     const uint16_t elements = VL_ / esize;
-    // If not pattern then same as ALL
-    if (operandStr.find(",") == std::string::npos) return elements;
+    const std::vector<std::string> patterns = {
```

Comment:
What happens when we get a pattern we don't support?

Reply:
All possible predicate patterns for armv8.4 and armv9.2 are included in this vector. However, if a pattern that wasn't supported was used then this function will treat it as an `all` (i.e. same as no pattern).

Reply:
Is it worth having a debug assert clause to avoid issues with unsupported patterns causing functional errors?

Reply:
So if we found a pattern in the operandStr that doesn't have an associated clause in the if-else structure below, then raise trigger an `assert` clause? It would be better if this were baked into the runtime (i.e. raise an NYI exec with some sort of extra clause to say it's the pattern that's not simply rather than the opcode). Additionally, how much work is it to have an else if clause for each pattern?

Reply:
I’m not sure I fully understand. For each pattern in the vector there is an associated if clause. Are you saying we should have a default clause to catch any other patterns? Or are you saying we should catch a pattern that isn’t in the patterns vector? As if the latter is the case I’m not sure it is possible

Reply:
Ah, some of the lines were omitted from the preview. Didn't see that all patterns have been supported. If all the patterns within the string vector contain all patterns supported under SVE then we should be good regarding functional issue due to lacking pattern support right?

Reply:
Yeah, the only case that a pattern is not recognised is if more are introduced in a new ISA release

Reply:
Then maybe a default case which raises a Debug assert would be worth it in those cases (i.e. just above the return 0 line)

Reply:
something like `assert(false && "Unknown predicate pattern")`

Reply:
Can do yes. Although I don’t think anything would trigger it. I should have clarified that the pattern string is initialised to “all” to accommodate for the fact that the pattern is optional, and so if no pattern in the vector is seen it assumes this is the case

Reply:
Ah I see this issue now. We may have to just keep a close eye on this then and update the vector as needed.

Reply:
I'll mark this as resolved but we need to remember this as a possible future issue.